### ISSUE TITLE
Unable to request new process from forkserver

### ISSUE THREAD

**Issue by cjb** [uds-se/AFLplusplus#1]
Hi folks, thanks for writing this! I'm seeing this error whenever I try to use your AFLplusplus fork. I've tried different distros (Ubuntu, Arch) and changing the mutator library or the harness binary, to no effect.  Any ideas?  I can build upstream aflplusplus on the same machine and run it without any issues.

```
 λ AFL_CUSTOM_MUTATOR_LIBRARY=$WORKDIR/FormatFuzzer/jpg.so ./afl-fuzz -S ffmut -i testcases/images/jpeg -o jpg -x dictionaries/jpeg.dict -- $WORKDIR/libjpeg-turbo/build/djpeg @@
afl-fuzz++2.59d based on afl by Michal Zalewski and a big online community
[+] afl++ is maintained by Marc "van Hauser" Heuse, Heiko "hexcoder" Eissfeldt and Andrea Fioraldi
[+] afl++ is open source, get it at https://github.com/vanhauser-thc/AFLplusplus
[+] Power schedules from github.com/mboehme/aflfast
[+] Python Mutator and llvm_mode whitelisting from github.com/choller/afl
[+] afl-tmin fork server patch from github.com/nccgroup/TriforceAFL
[+] MOpt Mutator from github.com/puppet-meteor/MOpt-AFL
[*] Getting to work...
[+] Using exploration-based constant power schedule (EXPLORE)
[+] You have 8 CPU cores and 1 runnable tasks (utilization: 12%).
[+] Try parallel jobs - see /usr/local/share/doc/afl/parallel_fuzzing.txt.
[*] Checking CPU core loadout...
[+] Found a free CPU core, binding to #0.
[*] Checking core_pattern...
[!] WARNING: Could not check CPU scaling governor
[*] Loading custom mutator library from '/home/cjb/fuzz/formatfuzzer/FormatFuzzer/jpg.so'...
mutation_infop 0x7ffff7d83780
[+] Custom mutator installed successfully.
[*] Setting up output directories...
[*] Scanning 'testcases/images/jpeg'...
[+] No auto-generated dictionary tokens to reuse.
[*] Creating hard links for all input files...
[*] Loading extra dictionary from 'dictionaries/jpeg.dict' (level 0)...
[+] Loaded 14 extra tokens, size range 2 B to 5 B.
[*] Validating target binary...
[*] Attempting dry run with 'id:000000,time:0,orig:not_kitty.jpg'...
[*] Spinning up the fork server...
[+] All right - fork server is up.

[-]  SYSTEM ERROR : Unable to request new process from fork server (OOM?)
    Stop location : run_target(), src/afl-fuzz-run.c:145
       OS message : Broken pipe

```

**Comment by RafaelTupynamba** [comment]
Hey, I just tried setting up everything from scratch and running the exact command you used (djpeg) on a server here (running Debian 10) and it worked without a problem.
Is it possible that you're running out of memory (OOM)?
You could also try with the exact version 2.60c of AFL++ that we are based on (commit 842cd9dec3c4c83d660d96dcdb3f5cf0c6e6f4fb) and see if that works on your system.

**Comment by cjb** [comment]
Thank you very much @RafaelTupynamba, I really appreciate the response!

I tried to replicate your setup exactly with a Debian 10 VM of my own, and it works now.  Previously I was trying on Ubuntu 21.10 in a VM, and Arch on a physical machine with plenty of RAM, and got the same error in both places.  So I might have just been doing something wrong before that I stopped doing here, or there might be a problem with those newer distros.  I'll close the issue, and update it in future if I learn more about which case it might have been.  Thanks again!
